packages/central-server/src/apiV2/getCountryAccessList.js

```diff
@@ -18,29 +18,34 @@ const mapRequestsToEntities = (requestedEntities, projectCodeById) => {
 };
 
 export async function getCountryAccessList(req, res, next) {
-  const { userId, models } = req;
+  const { userId, models, query, accessPolicy } = req;
+  const { projectCode } = query;
+  if (!projectCode) throw new Error('No project code provided');
 
   try {
-    const countries = await models.entity.find({ type: 'country' });
-    const entityPermissions = await models.userEntityPermission.find({ user_id: userId });
-    const permittedEntityIds = new Set(entityPermissions.map(p => p.entity_id));
-    const accessRequests = await models.accessRequest.find({
-      user_id: userId,
-      processed_date: null,
-    });
-    const projects = await models.project.find();
-    const projectCodeById = reduceToDictionary(projects, 'id', 'code');
-    const entityRequests = mapRequestsToEntities(accessRequests, projectCodeById);
+    const project = await models.project.findOne({ code: projectCode });
+    if (!project) throw new Error(`No project found with code ‘${projectCode}’`);
+
+    const countries = await project.countries();
+    const countriesPendingAccess = (
+      await models.accessRequest.find({
+        user_id: userId,
+        project_id: project.id,
+        approved: null,
```

Comment:
Just realised that the old code looked for `processed_date IS NULL` to mean “pending”

This made more sense in my head, but it was an off-the-cuff decision. If this is somehow not preferred or wrong, let me know